amgut/db/patches/0017.sql

```diff
@@ -0,0 +1,195 @@
+
+-- Assign barcodes to American Gut and Sleep Study
+CREATE TEMP TABLE hold_table (
+    barcode varchar,
+    project_id bigint
+);
+
+pid := (SELECT project_id FROM barcodes.project WHERE project = 'Sleep Study');
+
+INSERT INTO hold_table (project_id, barcode)
+    SELECT pid, barcode FROM (
+        SELECT barcode FROM ag.ag_kit_barcodes
+        JOIN ag.ag_kit USING (ag_kit_id)
+        WHERE lower(supplied_kit_id) LIKE '%slp\_%'
+        UNION
+        SELECT barcode from ag.ag_handout_barcodes
+        JOIN ag.ag_handout_kits USING (kit_id)
+        WHERE lower(kit_id) LIKE '%slp\_%') AS b;
+
+INSERT INTO hold_table (project_id, barcode)
+    SELECT ag_pid, barcode FROM hold_table;
+
+INSERT INTO barcodes.project_barcode (project_id, barcode)
+    SELECT project_id, barcode FROM hold_table
+    WHERE (project_id, barcode) NOT IN
+        (SELECT project_id, barcode FROM barcodes.project_barcode);
+
+DROP TABLE hold_table;
+
+-- Create ASD project
+INSERT INTO project (project_id, project)
+    SELECT max(project_id)+1, 'Autism Spectrum Disorder'
+    FROM project;
+
+pid := (SELECT project_id FROM barcodes.project WHERE project = 'Autism Spectrum Disorder');
+
+-- Assign all ASD kits to ASD project and American Gut
+CREATE TEMP TABLE hold_table (
+    barcode varchar,
+    project_id bigint
+);
+
+INSERT INTO hold_table (project_id, barcode)
+    SELECT pid, barcode FROM (
+        SELECT barcode FROM ag.ag_kit_barcodes
+        JOIN ag.ag_kit USING (ag_kit_id)
+        WHERE lower(supplied_kit_id) LIKE '%asd\_%'
+        OR lower(supplied_kit_id) LIKE '%asdp\_%'
+        UNION
+        SELECT barcode from ag.ag_handout_barcodes
+        JOIN ag.ag_handout_kits USING (kit_id)
+        WHERE lower(kit_id) LIKE '%asd\_%'
+        OR lower(kit_id) LIKE '%asdp\_%') AS b;
+
+INSERT INTO hold_table (project_id, barcode)
+    SELECT ag_pid, barcode FROM hold_table;
+
+INSERT INTO barcodes.project_barcode (project_id, barcode)
+    SELECT project_id, barcode FROM hold_table
+    WHERE (project_id, barcode) NOT IN
+        (SELECT project_id, barcode FROM barcodes.project_barcode);
+
+DROP TABLE hold_table;
+
+-- Unassign from American gut and assign to 'Amnon Oral Timeseries'
+CREATE TEMP TABLE hold_table (
+    barcode varchar,
+    project_id bigint
+);
+
+INSERT INTO project (project_id, project)
+    SELECT max(project_id)+1, 'Amnon Oral Timeseries' FROM project;
+
+pid := (SELECT project_id FROM barcodes.project WHERE project = 'Amnon Oral Timeseries');
+
+INSERT INTO hold_table (project_id, barcode)
+    SELECT pid, barcode FROM (
+        SELECT barcode FROM ag.ag_kit_barcodes
+        JOIN ag.ag_kit USING (ag_kit_id)
+        WHERE lower(supplied_kit_id) LIKE '%mts\_%'
+        OR lower(supplied_kit_id) LIKE '%osc\_%'
+        OR lower(supplied_kit_id) LIKE '%pulse\_%'
+        UNION
+        SELECT barcode from ag.ag_handout_barcodes
+        JOIN ag.ag_handout_kits USING (kit_id)
+        WHERE lower(kit_id) LIKE '%mts\_%'
```

Comment:
on all of these, the kit IDs are `lower`ed, why? Are the prefixes not stable in case?